lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2134,37 +2134,51 @@ namespace Js
     Var JavascriptString::ToCaseCore(JavascriptString* pThis, ToCase toCase)
     {
         charcount_t count = pThis->GetLength();
+        charcount_t countToCase = count;
 
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* inStrLim = inStr + count;
+        const wchar_t* i = inStr;
+
+        // Try to find out the chars that do not need casing
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim) {
+                if (*i >= 224 || (*i <= 118 && *i >= 97)) break;
+                i++;
+            }
+        } else {
+            Assert(toCase == ToLower);
+            while (i < inStrLim) {
+                if (*i >= 257 || (*i <= 204 && *i >= 192) || (*i <= 90 && *i >= 65)) break;
```

Comment:
It seems odd to include only 0x100 considering that a lot of that range might also be eligible. Otherwise keep it to strictly ASCII, or include a reference to a Unicode standard with rationale for the bounds you pick.

See: https://en.wikibooks.org/wiki/Unicode/Character_reference/0000-0FFF